### DISCUSSION TITLE
What does the warning `Unused co-located fragment definition(s)` mean?

### DISCUSSION THREAD

**Discussion by valtism** [0no-co/gql.tada#336] [Q&A]
I've been trying to follow best practices for fragments, and have extracted a fragment I use in a file called `Dashboard.tsx`

```
components
├ Dashboards.tsx <- query that uses fragment to fetch
└ Dashboard.tsx <- component that consumes fragment data
```

When I tried to run this in next, I get an error saying it can't use a fragment before it is defined, so I had to move the fragment from `Dashboard.tsx` to `Dashboards.tsx`

```
components
├ Dashboards.tsx <- both fragment and query live here now
└ Dashboard.tsx
```

Now, whenever I try and import something (like the dashboard mutation, that also lives in Dashboards), I get a warning saying `Unused co-located fragment definition(s) "DashboardFragment" in "@/src/components/Dashboards" ts(52003)`

Why is this happening, and what does this error mean?

**Comment by kitten** [comment]
Your fragment for `Dashboard.tsx`, which presumably contains data that `Dashboard` depends on lives in `Dashboards`, which violates this check and hence triggers the warning.

There's a docs page on this topic: https://gql-tada.0no.co/guides/fragment-colocation
